src/core/thread/mle_router.cpp

```diff
@@ -4339,6 +4339,8 @@ Error MleRouter::GetMaxChildTimeout(uint32_t &aTimeout) const
 
     VerifyOrExit(IsRouterOrLeader(), error = kErrorInvalidState);
 
+    aTimeout = 0;
```

Comment:
nit: move before `VerifyOrExit` ?

Reply:
I agree. Actually I prefer to leave the value untouched when exiting with errors. But current solution is worse because it's not consistent, and fixing that will add significant complexity. I'll move it before `VerifyOrExit` to make it consistent.